### DISCUSSION TITLE
SSO Issue when embedded Open WebUI in webpage or microsoft teams

### DISCUSSION THREAD

**Discussion by julienlebutte** [open-webui/open-webui#7770] [Issues]
## Installation Method

Kubernetees or Docker (Same issue)

## Environment

- **Open WebUI Version:** v0.4.8
- **Ollama (if applicable):** 0.5.1

- **Operating System:** Windows 10
- **Browser (if applicable):** Chrome or Teams

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the browser console logs.
- [X] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Open WebUI is installed on a specific container with Ollama.
Microsoft SSO is configured and working well on browser.
Open WebUI is behind a F5 with only one node.

## Actual Behavior:

When i access the Open WebUI website, SSO authentication is working well.
When i access the Open WebUI website include in an iframe website, SSO is not working.
When i access the Open WebUI website include in a Microsoft Teams application, SSO is not working.

## Description

**Bug Summary:**

Based on the SSO scheme, call to EntraID is working, i see the authentication on the identity provider.
On the network trace, the callback url give an error : 400 bad request

## Reproduction Details

**Steps to Reproduce:**
Configure Microsoft authetication on Open WEbUi
Integrate Open WebUI site on any other site with an iframe (SharePoint, Wordpress, Microsoft Teams, ....)
Try to log in

## Logs and Screenshots

**Browser Console Logs:**

<img width="1084" alt="openwebui1" src="https://github.com/user-attachments/assets/69174b3e-ef51-439b-9bfe-f5b74dcd5e10">

**Docker Container Logs:**

WARNI [open_webui.utils.oauth] OAuth callback error: mismatching_state: CSRF Warning! State not equal in request and response.
INFO:     10.42.0.1:0 - "GET /oauth/microsoft/callback?code=1.ASEAUm_EtlxiS0GsosSk8Oe9IGApLSeqYV5Ms0Et-YbXf8AYAS0hAA.AgABBAIAAADW6jl31mB3T7ugrWTT8pFeAwDs_wUA9P_jcxwySU6P0fMNbIPV7-jSPhpvX1VFR2DE5mK2bfGQKeN34e1z8tDf-ayN10--5cBq3ER77v7uaFsR9xdIEdcHd7YB0hZMUejCIQgEXOtiBlQkAtXZf0k0irIJL-w2zP_rhgVbISeOnqC6YQuUeYrqLoAAksnYgeSSc4JO4NmRU9Hy6OVjqFAArNfEL2ll-9svMpY7PcvpMVp1UdVTiMVfWT1pooaKPoe3qxR0bDzoJN_2XHHXoO-pdtOcGdY6RMyeQQattrDSZ_vbZo70t33R7uY2uUGyapKQBk7hlDDUPa-NfD_cbo5IGhkXdeaa4fjgyEpighFmQFThBm6csJ7i7fnvSPi_aKPnstGgCOaNEiK3OVnhq8qqBiAMEaT5-LtMB2qP20NdIKrzhuwpnX392YCl3SDjvIBLCzlenbzY0IoqMUdX6njzeUsvbYNKrfhGZ0iCLFyLx1bwiL8qhKSDtgA-XXXXXXXXX-9hSFHdKa7bJ8_YxCMvrPXpLSLEwNNsXfmSdRk4hI7S5JISyiDNMPst8jq7cZO3SeNmhDqF8Uh0JP-Ty7l50iw64dk-ypnLzpQUqmR0cAY0O3u62EwnvzZOt6CSErP_H7Q-r2mcwl_ObE5tL2x44WiQhttP-sJ6bFJrnw05yTIvJnJ01YFa-ERvg-XXXXXXX-tinUzQb2Q3eAbHn4gBkMMHIgRoZ5IBYCvkHnVfGeFY1u0BqW6d_NkFZmVZ78XDgMbjGw8&state=xb57MICcVFdlOjqJyD9lyb2t8Uq3MX&session_state=XXXXXXXXXXX HTTP/1.1" 400 Bad Request
INFO:     connection closed

**Comment by Rohan-Hazari** [comment]
Any update ? I'm facing the same issue when I access through iframe and specifically the Microsoft provider, if I do it on the website it works fine

**Comment by dawnmarch** [comment]
same issue

**Comment by arunbugkiller** [comment]
Same issue

**Comment by dsjath** [comment]
I am facing the same issue

**Comment by arunbugkiller** [comment]
@dsjath @julienlebutte, what environment variables are you using, please make sure you are configuring the below environment variables, for me doing so worked 

ENABLE_OAUTH_SIGNUP=true 
WEBUI_SESSION_COOKIE_SAME_SITE=lax 
WEBUI_AUTH_COOKIE_SAME_SITE=lax 
ENABLE_LOGIN_FORM=True 
OAUTH_MERGE_ACCOUNTS_BY_EMAIL=true

**Comment by arunbugkiller** [comment]
@julienlebutte - I am facing this issue in latest version 0.6.0. Were you able to resolve this 400 error, I am getting the same as yours on MS Teams

2025-04-05 05:09:36.346 | WARNING  | open_webui.utils.oauth:handle_callback:237 - OAuth callback error: mismatching_state: CSRF Warning! State not equal in request and response. - {}
2025-04-05 05:09:36.346 | INFO     | uvicorn.protocols.http.httptools_impl:send:476 - 18.191.116.100:0 - "GET /oauth/microsoft/callback?code=1.AS0AajkibDrW_k2lAUxCdjaHVVYliqvgoihNlRwZk50PG64tACstAA.AgABBAIAAABVrSpeuWamRam2jAF1XRQEAwDs_wUA9P_Hi8NxWjvoyb2qDblgMDs5M6RH94024LCoRIHd5NEdB80p44uVxRvFMLTo9ydnN7SFL8Wc4CssbBa3pJMMAWxVR4_QfGFts0GFMeLpKS7J0tyV2Bgs8ANLUQHEKt4YtRJMq7rrfKtaT6HO0zPHVQjvRokNPImYXlBnlI0AIXmR-cT1DMwuRV0-tQAmj9Ut3G7xw3wHHzZuMujwXYpBRfB27AcG6r83CuQ4x350hALYFuH8vjmAlWObSfEje9HTxOFV3XBg8WTLYEpJHSwTOxpbdDZIR3vxWN6Lz-o-tIPaVbC3GUp1bVXtoQm3QeM2iSRtdK_d7VnrHHMxOZ6Sz2UF4t26wZlyhWwgR_fI4zikiU7MyR5n2Y8xAMw0lwl_lNQgvvoGraGq9FA8UFm8Q87j7IHq4UctXiSWThpoVqdCB_qH_h295V-cF6r_dCyUxI3n4SquiVbzxQRUOtE07Bo_ivktLum170fnhuG9y_bD-xAOT_F8pRP0wuAE873AiZk_IHudmhQdyFHH9erpNzIT_sFlMRcRgdOVzGezTjyjAV1plVX1QbSVvK1_lQjIXmZql_xB-2p8U3kGlXh_F8xcWjzi4Enh-Tx5JwR34h_OWiP-_fPnxmEY9FPdy_EGWbbnK0opByz2jOGS-63dJrvIBrClgdLc0iJUn8skfxSSxgMG5BmLJ6fphCYSIaZgbu0AMtZORZ3ajB2cVr8VbYEjk48t92bi1b7knViYSCo90xfljXBEEp5HajGNZLsiFldosxboNB8BpktsWYLQy4DQwGxU8dAlxgLmW6u1qNghKVjzTKeS_W5eePc8p_SjUdSFuJgtELaiBPTqAd5TVPjNLmvFA5LXl7nmWg2EuSYe1JSYvXPiKBFn1R5V_7yP6lb_96GKQBbcdrHnOFnI0G0Cn4En-YhvrRqk-OXXqQmRaGBF9WP4yB5dRq9_0DXJCAjeUV3InWfD97jUVfk&state=GI5eyoy7hHgYXoIubYcaLy31taWwOu&session_state=00206129-e73f-f073-f99c-9fe102fec434 HTTP/1.1" 400 - {}

**Comment by asiedler** [comment]
Hi together,

i was able to add openwebui via ms teams app.
For that i got the added the    domains as validDomains;
    "login.microsoftonline.com",
    "xxx.sharepoint.com"

i also created a new Azure Client App registration and added the WebApplicationInfo in the manifest
```
"webApplicationInfo": {
    "id": "ClientID",
    "resource": "api://OPENWEBUIURL/ClientID"
  }
```
  
 In the azure app reg i have just User.Read permission and exposed an API and defined the application ID URI like in the example above. There i added a new scope like access_as_user
 
 We host our Openwebui in a docker container inside of an Azure App Service. There we enable authentication etc. 
 In the Identity provider settings i added the created scope of the ms teams app registration to the allowed token audiences and enable the client requrements so that tis app can talk to the app services.
 
 In the openwebui added the CONTENT_SECURITY_POLICY and added the urls from microsoft:
 frame-ancestors 'self' *.microsoft.com *.sharepoint.com *.onedrive.com *.teams.microsoft.com gpt.pfeifer-langen.com *.cloud.microsoft
 
 I did a lot of try and error but i think these settings will get it done

**Comment by scalenow** [comment]
similar issue https://github.com/open-webui/open-webui/discussions/16122

**Comment by scalenow** [comment]
@arunbugkiller tried this sorry did not work   ENABLE_OAUTH_SIGNUP=true
WEBUI_SESSION_COOKIE_SAME_SITE=lax
WEBUI_AUTH_COOKIE_SAME_SITE=lax
ENABLE_LOGIN_FORM=True
OAUTH_MERGE_ACCOUNTS_BY_EMAIL=true
